build/test/generate/expected_outputs_test.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:build/build.dart';
+import 'package:build_test/build_test.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
Lets also add some tests validating the behavior when there is a capture group at the start or end of an input string, as well as when the match is potentially ambiguous, to demonstrate the eagerness (or not) of the matching.